src/components/Button/ButtonWrapper.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { ReactNode, useMemo } from 'react'
+import styled, { css } from 'styled-components'
+
+import { Theme, useTheme } from '../../hooks/useTheme'
+
+import { Variant } from './types'
+
+type Props = {
+  size: 'default' | 's'
+  square: boolean
+  wide: boolean
+  variant: Variant
+  className: string
+  children: ReactNode
+  isAnchor?: boolean
+}
+type StyleProps = Pick<Props, 'wide' | 'variant'> & { themes: Theme }
+
+export function ButtonWrapper<T extends Props>({ size, square, className, isAnchor, ...props }: T) {
+  const theme = useTheme()
+  const buttonClassName = useMemo(
+    () => `${size} ${className} ${square ? 'square' : ''}`,
+    [className, size, square],
+  )
+
+  return isAnchor ? (
+    <Anchor {...props} className={buttonClassName} themes={theme} />
+  ) : (
+    <Button {...props} className={buttonClassName} themes={theme} />
+  )
+}
+
+const baseStyles = css<StyleProps>(({ wide, themes }) => {
+  const { border, fontSize, leading, radius, shadow, spacingByChar } = themes
+
+  return css`
+    box-sizing: border-box;
+    cursor: pointer;
+    display: inline-flex;
+    justify-content: center;
+    align-items: center;
+    gap: ${spacingByChar(0.5)};
+    text-align: center;
+    white-space: nowrap;
+    border-radius: ${radius.m};
+
+    /* ボタンの高さを合わせるために指定 */
+    border: ${border.lineWidth} ${border.lineStyle} transparent;
+    padding: ${spacingByChar(0.75)} ${spacingByChar(1)};
+    font-family: inherit;
+    font-size: ${fontSize.M};
+    font-weight: bold;
+    line-height: ${leading.NONE};
+    ${wide && 'width: 100%;'}
+
+    &.square {
+      padding: ${spacingByChar(0.75)};
+    }
+
+    &.s {
+      padding: ${spacingByChar(0.5)};
+      font-size: ${fontSize.S};
+
+      /* ボタンラベルの line-height を 0 にしたため、高さを担保する */
+      min-height: calc(${fontSize.S} + ${spacingByChar(1)} + (${border.lineWidth} * 2));
+    }
+
+    &:focus {
+      ${shadow.focusIndicatorStyles}
+    }
+
+    /* baseline より下の leading などの余白を埋める */
+    .smarthr-ui-Icon,
+    svg {
+      display: block;
+    }
+  `
+})
+
+const Button = styled.button<StyleProps>(({ variant, themes }) => {
+  const styles = variantStyles(variant, themes)
+  return css`
+    ${baseStyles}
+    ${styles.default}
+
+    &:focus,
+    &:hover {
+      ${styles.focus}
+    }
+    &[disabled] {
+      cursor: not-allowed;
+      ${styles.disabled}
+
+      /* alpha color を使用しているので、背景色と干渉させない */
+        background-clip: padding-box;
+    }
+  `
+})
+
+const Anchor = styled.a<StyleProps>(({ variant, themes }) => {
+  const styles = variantStyles(variant, themes)
+  return css`
+    ${baseStyles}
+    ${styles.default}
+    text-decoration: none;
+
+    &:focus,
```

Comment:
`:focus-visible` でお願いします。

```suggestion
    &:focus-visible,
```